File: C++/2024-12-30/148653.cpp
#include <string>
#include <vector>
using namespace std;

int solution(int storey) 
{
    int answer = 0;

	while (storey > 0)
	{
		int floor = storey % 10;
		int next = storey / 10 % 10;

		if (floor > 5 || (floor >= 5 && next >= 5))
		{
			answer += 10 - floor;
			storey += 10 - floor;
		}
		else
		{
			answer += floor;
		}

		storey /= 10;
	}

    return answer;
}

File: JAVA/148653.java
public class Main {
    public static void main(String[] args) {
    }
}

class solution
{
    public int solution(int storey)
    {

        int result = 0; 
        int carry = 0;
    
  
        String[] digits = Integer.toString(storey).split("");
    

        for (int i = digits.length - 1; i >= 0; i--) {
            int digit = Integer.parseInt(digits[i]); 
            digit += carry; 
    
            if (digit > 5 || (digit == 5 && i > 0 && Integer.parseInt(digits[i - 1]) >= 5)) {
                result += 10 - digit;
                carry = 1;
            }
            else
            {
                result += digit;
                carry = 0; 
            }
        }
    

        if (carry > 0) {
            result += carry;
        }
    
        return result;
    }
}
